transport/transport.go

```diff
@@ -330,12 +330,20 @@ func NewServerTransport(protocol string, conn net.Conn, maxStreams uint32, authI
 	return newHTTP2Server(conn, maxStreams, authInfo)
 }
 
+// DefaultDialerFn is the default dialer.
+func DefaultDialerFn(addr string, timeout time.Duration, cancel <-chan struct{}) (net.Conn, error) {
```

Comment:
only non-test usage (i.e., newHTTP2Client) is in transport package -- you do not need to export it. 

Transport package is bit in the grey area -- we may allow some proxy to use it directly.